src/modules/ledger/services/LedgerService.ts

```diff
@@ -30,27 +27,43 @@ export class LedgerService {
   private _poolHandle?: PoolHandle
   private authorAgreement?: AuthorAgreement | null
   private indyIssuer: IndyIssuerService
-
-  public constructor(@inject(Symbols.Wallet) wallet: Wallet, agentConfig: AgentConfig, indyIssuer: IndyIssuerService) {
+  private agentConfig: AgentConfig
+  private fileSystem: FileSystem
+
+  public constructor(
+    @inject(Symbols.Wallet) wallet: Wallet,
+    agentConfig: AgentConfig,
+    indyIssuer: IndyIssuerService,
+    @inject(Symbols.FileSystem) fileSystem: FileSystem
+  ) {
     this.wallet = wallet
+    this.agentConfig = agentConfig
     this.indy = agentConfig.indy
     this.logger = agentConfig.logger
     this.indyIssuer = indyIssuer
+    this.fileSystem = fileSystem
   }
 
-  private get poolHandle() {
+  private async getPoolHandle() {
     if (!this._poolHandle) {
-      throw new Error('Pool has not been initialized yet.')
+      return this.connect()
     }
 
     return this._poolHandle
   }
 
-  public async connect(poolName: string, poolConfig: LedgerConnectOptions) {
-    this.logger.debug(`Connecting to ledger pool '${poolName}'`, poolConfig)
+  private async connect() {
```

Comment:
I think a developer should be able to make the connection if they decide. In our use case, we found some problems when connecting to the staging from the mobile app. Sometimes it takes tens of seconds. Therefore it's convenient to separate the connection to the ledger from the wallet initialization but start making the connection right after and don't wait until the connection is needed.

Reply:
I agree. Updated